### PR TITLE
[To Main] Feature - DESENG-667: Add engagement configuration summary

### PR DESCRIPTION
**Pull request by NatSquared**
Issue #: https://citz-gdx.atlassian.net/browse/DESENG-667

**Description of changes:**
- **Feature** Add engagement configuration summary [🎟️ DESENG-667](https://citz-gdx.atlassian.net/browse/DESENG-667)

  - Added a tabbed layout to the new engagement view page
  - Added a new "Configuration" tab to display the engagement's configuration details
    - The other tabs are blank for now, but will be filled in future tickets
  - The configuration tab allows navigating back to the engagement configuration page
  - Improvements to how engagements and related resources are fetched and saved

**User Guide update ticket (if applicable):**

### CODE REVIEW

File: met-web/src/components/engagement/admin/config/EngagementUpdateAction.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        console.error('Error updating engagement slug', e);
+    }
+
+    const currentTeamMembers = await getTeamMembers({ engagement_id: engagementId });
+    const users = formData.getAll('users') as string[];
+    const usersSet = new Set(users);
+
+    try {
+        // Process deactivated users for reinstatement (and active users for revocation)
+        // Caution - headaches ahead! There is a big difference between user_id and user.external_id
```
Comment (Baelx) [inline met-web/src/components/engagement/admin/config/EngagementUpdateAction.tsx:37]: Thanks for helping future devs avoid headaches!

File: met-web/src/components/engagement/admin/view/ConfigSummary.tsx [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+                                            color: 'white',
+                                            width: '32px',
+                                            height: '32px',
+                                            '&:hover': {
+                                                backgroundColor: 'primary.main',
+                                            },
+                                            ...globalFocusVisible,
+                                            display: 'inline-block',
+                                            marginRight: '0.5rem',
+                                        }}
```
Comment (Baelx) [inline met-web/src/components/engagement/admin/view/ConfigSummary.tsx:61]:
```suggestion
                                        }}
                                        aria-label="Press enter to copy engagement URL to clipboard"
```

File: met-web/src/routes/AuthenticatedRoutes.tsx [hunk 3]
```diff
@@ -79,34 +81,39 @@ const AuthenticatedRoutes = () => {
...
                         element={<EngagementCreationWizard />}
-                        loader={languageLoader}
                     />
                 </Route>
                 <Route
                     path=":engagementId"
                     id="single-engagement"
                     errorElement={<NotFound />}
                     loader={engagementLoader}
+                    handle={{
```
Comment (Baelx) [inline met-web/src/routes/AuthenticatedRoutes.tsx:91]: Thanks for adding this other crumb. Do you know why we have both this route and "view" that point to the view page?

**Review by Baelx — CHANGES_REQUESTED** [review]
Thanks for this quick work Nat! Could you please just review that screen reader users will have an okay time with the tooltip button for copying the URL? Ideally they'd get all the same feedback as sighted users but if not we can just make sure the button instructions are good.

Also, maybe it'd be a good idea for us all to start communicating more about adding net-new components and changing app infrastructure. I'm trying to be mindful of managing complexity with this app. Could you maybe see if you could synthesize ConfigWizard and CreationWizard? I won't be insistent on it since you know the code best but we need to try to be economical whenever we can.

File: met-web/src/components/engagement/admin/config/wizard/ConfigWizard.tsx [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+                        xs={11}
+                        container
+                        direction="row"
+                        justifyContent="flex-start"
+                        alignItems="space-between"
+                        rowSpacing={1}
+                    >
+                        <Grid container direction="row" item xs={12}>
+                            <Grid xs={12}>
+                                <Header2 sx={{ mb: 0 }}>We're just looking over your configuration.</Header2>
```
[the remark this answers is not in the document]
Comment (NatSquared) [reply met-web/src/components/engagement/admin/config/wizard/ConfigWizard.tsx:134]: I got the language from here: https://www.figma.com/design/UXbho1MhS3DUYcBHHffzG0/MET-Engagement-Authoring-New-Designs?node-id=3875-13771&m=dev
Let me know what you'd like it changed to
This message would come up while server-side (form) validation is being performed; after field validation is done on the client

**Review by Baelx — APPROVED** [review]

**Pull request merged**